WcaOnRails/app/helpers/competitions_helper.rb

```diff
@@ -1,2 +1,25 @@
 module CompetitionsHelper
+  def competition_message_for_user(competition, user)
+    message = ""
+    if registration = competition.registrations.find_by_user_id(user.id)
+      if registration.accepted?
+        message = "You are registered. "
+      else
+        message = "You are currently on the waiting list. "
+      end
+    end
+    if competition.isConfirmed
+      if competition.showAtAll
+        message += "This competition is confirmed and visible."
+      else
+        message += "This competition is confirmed but not visible."
+      end
```

Comment:
Actually appearance of this code depends on what we consider to be readable enough, I can imagine this to be that short:
```ruby
  def competition_message_for_user(competition, user)
    message = ""
    if registration = competition.registrations.find_by_user_id(user.id)
      message = registration.accepted? ? "You are registered. "
                                       : "You are currently on the waiting list. "
    end
    confirmed, visible = [competition.isConfirmed, competition.isVisible]
    message += "This competition is #{'not ' unless confirmed}confirmed #{confirmed == visible ? 'and' : 'but'} #{'not ' unless visible}visible."
  end
```
(Maybe `if !` would sound better than `unless` in this particular case?)

What's the *golden mean* in your opinion?